assets/js/modules/analytics/datastore/profiles.js

```diff
@@ -146,25 +188,33 @@ export const actions = {
 		};
 	},
 
-	receiveProfilesCompleted( accountID, propertyID ) {
-		invariant( accountID, 'accountID is required.' );
-		invariant( propertyID, 'propertyID is required.' );
+	/**
+	 * Sets a profile based on given account and property IDs.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {string} propertyID Property ID.
+	 */
+	*setProfileForProperty( propertyID ) {
+		invariant( isValidPropertySelection( propertyID ), 'A valid property selection is required to set profile.' );
+		const registry = yield Data.commonActions.getRegistry();
 
-		return {
-			payload: { accountID, propertyID },
-			type: RECEIVE_PROFILES_COMPLETED,
-		};
-	},
+		if ( PROPERTY_CREATE === propertyID ) {
+			registry.dispatch( STORE_NAME ).setProfileID( PROFILE_CREATE );
+			return;
+		}
+		// Clear any profile ID selection in the case that selection falls to the getProfiles resolver.
+		registry.dispatch( STORE_NAME ).setProfileID( '' );
```

Comment:
Once again: I wonder about our use of empty string as a kind of magic constant. Could we use `null` here instead?